[fastlane] Delete IDFA declarations for Apple App Store submission

fr0l opened the pull request:
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue following this format:
Resolves #999999
-->

Resolves #21998

Seems Seems that the IDFA-related APIs were [deprecated in v1.5](https://developer.apple.com/documentation/appstoreconnectapi/app_store_connect_api_release_notes/app_store_connect_api_1_5_release_notes) by Apple and finally removed.

> The Advertising Identifier (IDFA) Declarations resource and all its associated endpoints and objects are now deprecated, including: Create an IDFA Declaration, Modify an IDFA Declaration, Delete an IDFA Declaration, Read the IDFA Declaration Information of an App Store Version, IdfaDeclaration, IdfaDeclarationCreateRequest, IdfaDeclarationUpdateRequest, and IdfaDeclarationResponse.

The PR is just to delete the deprecated IDFA-related code.



the issue  #21998 itself can be mitigated by:

- a) setting `add_id_info_uses_idfa` to false
- b) deleting `add_id_info_uses_idfa` from `Deliverfile` / call to `deliver` method

the PR is just code removal


### Description

<!-- Describe your changes in detail, as well as how you tested them. -->

### Testing Steps

<!-- Optional: steps, commands, or code used to test your changes. -->
<!-- Providing these will reduce the time needed for testing and review by the fastlane team. -->

fr0l committed:
[fastlane] Delete IDFA declarations for Apple App Store submission

fr0l commented:
Here is a diff between [openapi.1.4.1.json](https://github.com/fastlane/fastlane/files/15193119/openapi.1.4.1.json) and [openapi.3.4.1.json](https://github.com/fastlane/fastlane/files/15193120/openapi.3.4.1.json)


`diff -u openapi.1.4.1.json openapi.3.4.1.json|colordiff|grep -i idfa`


```diff
-                  "idfaDeclaration",
-                  "usesIdfa",
-                  "idfaDeclaration",
-            "description": "the fields to include for returned resources of type idfaDeclarations",
-            "name": "fields[idfaDeclarations]",
-                  "idfaDeclaration",
-                  "usesIdfa",
-                  "idfaDeclaration",
-                  "usesIdfa",
-    "/v1/appStoreVersions/{id}/idfaDeclaration": {
-        "operationId": "appStoreVersions-idfaDeclaration-get_to_one_related",
-            "description": "the fields to include for returned resources of type idfaDeclarations",
-            "name": "fields[idfaDeclarations]",
-                  "$ref": "#/components/schemas/IdfaDeclarationResponse"
-                  "idfaDeclaration",
-                  "usesIdfa",
-                  "idfaDeclaration",
-                  "usesIdfa",
-            "description": "the fields to include for returned resources of type idfaDeclarations",
-            "name": "fields[idfaDeclarations]",
-                  "idfaDeclaration",
-                  "usesIdfa",
-                  "idfaDeclaration",
-                  "idfaDeclaration",
-                  "usesIdfa",
-                  "idfaDeclaration",
-                  "usesIdfa",
-                  "idfaDeclaration",
-                  "usesIdfa",
-    "/v1/idfaDeclarations": {
-        "operationId": "idfaDeclarations-create_instance",
-                "$ref": "#/components/schemas/IdfaDeclarationCreateRequest"
-          "description": "IdfaDeclaration representation",
-                  "$ref": "#/components/schemas/IdfaDeclarationResponse"
-            "description": "Single IdfaDeclaration"
-          "IdfaDeclarations"
-    "/v1/idfaDeclarations/{id}": {
-        "operationId": "idfaDeclarations-delete_instance",
-          "IdfaDeclarations"
-        "operationId": "idfaDeclarations-update_instance",
-                "$ref": "#/components/schemas/IdfaDeclarationUpdateRequest"
-          "description": "IdfaDeclaration representation",
-                  "$ref": "#/components/schemas/IdfaDeclarationResponse"
-            "description": "Single IdfaDeclaration"
-          "IdfaDeclarations"
-              "usesIdfa": {
-              "idfaDeclaration": {
-                          "idfaDeclarations"
-                  "usesIdfa": {
-                  "$ref": "#/components/schemas/IdfaDeclaration"
-                  "usesIdfa": {
-                  "$ref": "#/components/schemas/IdfaDeclaration"
-      "IdfaDeclaration": {
-              "idfaDeclarations"
-        "title": "IdfaDeclaration",
-      "IdfaDeclarationCreateRequest": {
-                  "idfaDeclarations"
-        "title": "IdfaDeclarationCreateRequest",
-      "IdfaDeclarationResponse": {
-            "$ref": "#/components/schemas/IdfaDeclaration"
-        "title": "IdfaDeclarationResponse",
-      "IdfaDeclarationUpdateRequest": {
-                  "idfaDeclarations"
-        "title": "IdfaDeclarationUpdateRequest",
```

AliSoftware commented on fastlane/lib/fastlane/actions/docs/upload_to_app_store.md.erb:
Thanks for remembering to also update the docs! 🙇

AliSoftware reviewed (commented):
Thanks for tackling this!

I know you said it's still in progress, but from a quick skim of the diff and what was deleted and cleaned up, this looks good and very promising 👍

AliSoftware commented:
In case you don't have access to our CircleCI logs to see the test failure, here's the logs of `rspec` failing:

```
Failures:

  1) Deliver::SubmitForReview submit app submit fails needs to set export_compliance_uses_encryption
     Failure/Error:
       expect do
         review_submitter.submit!(options)
       end.to raise_error("boom")

       expected Exception with "boom", got #<RSpec::Mocks::MockExpectationError: #<Double "app"> received unexpected message :get_in_progress_review_submission with ({:platform=>"IOS"})> with backtrace:
         # ./deliver/lib/deliver/submit_for_review.rb:33:in `create_review_submission'
         # ./deliver/lib/deliver/submit_for_review.rb:25:in `submit!'
         # ./deliver/spec/submit_for_review_spec.rb:80:in `block (5 levels) in <top (required)>'
         # ./deliver/spec/submit_for_review_spec.rb:79:in `block (4 levels) in <top (required)>'
         # ./.bundle/ruby/3.2.0/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
     # ./deliver/spec/submit_for_review_spec.rb:79:in `block (4 levels) in <top (required)>'
     # ./.bundle/ruby/3.2.0/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

EthanSK commented:
Is this going to be merged in soon? I can't submit with fastlane atm. I either get "the specified resource does not exist - The path provided does not match a defined resource type" when add_id_info_uses_idfa is true or  Use of Advertising Identifier (IDFA) is required to submit

AliSoftware commented:
@EthanSK  you can just remove `add_id_info_uses_idfa` from your call to `deliver` / `submit_to_app_store` in your Fastfile (or from your `Deliverfile`) in your project and you'll be able to submit again.

(This PR is only to clean up what will now be dead code, as people stop using that attribute in their `Fastfile` and we start ignoring it now that it shouldn't be used anymore.)

cooksimo commented:
> @EthanSK you can just remove `add_id_info_uses_idfa` from your call to `deliver` / `submit_to_app_store` in your Fastfile (or from your `Deliverfile`) in your project and you'll be able to submit again.
> 
> (This PR is only to clean up what will now be dead code, as people stop using that attribute in their `Fastfile` and we start ignoring it now that it shouldn't be used anymore.)

This doesn't work because fastlane bails out of the submit if the idfa stuff is missing. It worked for us last release, but now I assume something has changed and the idfa declaration is missing which trips this issue: deliver/lib/deliver/submit_for_review.rb:148

dave-perry commented:
> > @EthanSK you can just remove `add_id_info_uses_idfa` from your call to `deliver` / `submit_to_app_store` in your Fastfile (or from your `Deliverfile`) in your project and you'll be able to submit again.
> > (This PR is only to clean up what will now be dead code, as people stop using that attribute in their `Fastfile` and we start ignoring it now that it shouldn't be used anymore.)
> 
> This doesn't work because fastlane bails out of the submit if the idfa stuff is missing. It worked for us last release, but now I assume something has changed and the idfa declaration is missing which trips this issue: [deliver/lib/deliver/submit_for_review.rb:148](https://github.com/fastlane/fastlane/blob/e8fe2a941102b41acedb7e6596ccf1fccf46419d/deliver/lib/deliver/submit_for_review.rb#L148)

Having this exact same issue.

fr0l commented:
the PR should be good to go. I have performed several submissions with the IDFA deleted, and it seems OK so far

fr0l commented:
> > > @EthanSK you can just remove `add_id_info_uses_idfa` from your call to `deliver` / `submit_to_app_store` in your Fastfile (or from your `Deliverfile`) in your project and you'll be able to submit again.
> > > (This PR is only to clean up what will now be dead code, as people stop using that attribute in their `Fastfile` and we start ignoring it now that it shouldn't be used anymore.)
> > 
> > 
> > This doesn't work because fastlane bails out of the submit if the idfa stuff is missing. It worked for us last release, but now I assume something has changed and the idfa declaration is missing which trips this issue: [deliver/lib/deliver/submit_for_review.rb:148](https://github.com/fastlane/fastlane/blob/e8fe2a941102b41acedb7e6596ccf1fccf46419d/deliver/lib/deliver/submit_for_review.rb#L148)
> 
> Having this exact same issue.

Hey @dave-perry and @cooksimo !
Please could you try with current release of Fastlane `2.220.0` with using
```ruby
submission_information({
  ...
  add_id_info_uses_idfa: false,
  ...
})
```
in `Deliverfile`

I've just tried one submission and it seems to work

fr0l commented:
Also PR for docs update https://github.com/fastlane/docs/pull/1246

EthanSK commented:
> > > > @EthanSK you can just remove `add_id_info_uses_idfa` from your call to `deliver` / `submit_to_app_store` in your Fastfile (or from your `Deliverfile`) in your project and you'll be able to submit again.
> > > > (This PR is only to clean up what will now be dead code, as people stop using that attribute in their `Fastfile` and we start ignoring it now that it shouldn't be used anymore.)
> > > 
> > > 
> > > This doesn't work because fastlane bails out of the submit if the idfa stuff is missing. It worked for us last release, but now I assume something has changed and the idfa declaration is missing which trips this issue: [deliver/lib/deliver/submit_for_review.rb:148](https://github.com/fastlane/fastlane/blob/e8fe2a941102b41acedb7e6596ccf1fccf46419d/deliver/lib/deliver/submit_for_review.rb#L148)
> > 
> > 
> > Having this exact same issue.
> 
> Hey @dave-perry and @cooksimo ! Please could you try with current release of Fastlane `2.220.0` with using
> 
> ```ruby
> submission_information({
>   ...
>   add_id_info_uses_idfa: false,
>   ...
> })
> ```
> 
> in `Deliverfile`
> 
> (in order to unblock you)
> 
> I've just tried one submission and it seems to work
> 
> Just for example. This is what I see in logs:
> 
> ```
> DEBUG [2024-05-10 12:12:30.31]: Updating app store version for IDFA status of 'false'
> INFO  [12:12:30]: >> PATCH appStoreVersions/blah-blah-blah: {"data":{"type":"appStoreVersions","id":"blah-blah-blah","attributes":{"usesIdfa":false}}} 
> DEBUG [12:12:31]: << PATCH appStoreVersions/blah-blah-blah: 200 {"data"=>{"type"=>"appStoreVersions", "id"=>"blah-blah-blah", "attributes"=>{"platform"=>"IOS", "versionString"=>"2.354.0", "appStoreState"=>"PREPARE_FOR_SUBMISSION", "appVersionState"=>"PREPARE_FOR_SUBMISSION", "copyright"=>"© 2003-2024 Blah-Blah Limited", "reviewType"=>"APP_STORE", "releaseType"=>"MANUAL", "earliestReleaseDate"=>nil, "usesIdfa"=>false, "downloadable"=>true, "createdDate"=>"2024-05-08T03:43:35-07:00"}, "relationships"=>{"ageRatingDeclaration"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/ageRatingDeclaration", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/ageRatingDeclaration"}}, "appStoreVersionLocalizations"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/appStoreVersionLocalizations", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/appStoreVersionLocalizations"}}, "build"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/build", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/build"}}, "appStoreVersionPhasedRelease"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/appStoreVersionPhasedRelease", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/appStoreVersionPhasedRelease"}}, "gameCenterAppVersion"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/gameCenterAppVersion", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/gameCenterAppVersion"}}, "routingAppCoverage"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/routingAppCoverage", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/routingAppCoverage"}}, "appStoreReviewDetail"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/appStoreReviewDetail", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/appStoreReviewDetail"}}, "appStoreVersionSubmission"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/appStoreVersionSubmission", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/appStoreVersionSubmission"}}, "idfaDeclaration"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/idfaDeclaration", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/idfaDeclaration"}}, "appClipDefaultExperience"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/appClipDefaultExperience", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/appClipDefaultExperience"}}, "appStoreVersionExperiments"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/appStoreVersionExperiments", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/appStoreVersionExperiments"}}, "appStoreVersionExperimentsV2"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/appStoreVersionExperimentsV2", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/appStoreVersionExperimentsV2"}}, "customerReviews"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/customerReviews", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/customerReviews"}}, "alternativeDistributionPackage"=>{"links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/relationships/alternativeDistributionPackage", "related"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah/alternativeDistributionPackage"}}}, "links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah"}}, "links"=>{"self"=>"https://api.appstoreconnect.apple.com/v1/appStoreVersions/blah-blah-blah"}}
> DEBUG [2024-05-10 12:12:31.10]: Updated app store version for IDFA status of 'false'
> INFO [2024-05-10 12:12:31.10]: Successfully updated IDFA declarations on App Store Connect
> ```
> 
> I hope it helps

It seems to sometimes work and sometimes not. For me as well, the last submission went through without idfa but then it failed recently...

fr0l commented:
Feedback addressed @AliSoftware 
Please could you give it a second review. should be OK now.
Thank you

dave-perry commented:
> Hey @dave-perry and @cooksimo ! Please could you try with current release of _fastlane_ `2.220.0` with using
> 
> ```ruby
> submission_information({
>   ...
>   add_id_info_uses_idfa: false,
>   ...
> })
> ```
> 
> in `Deliverfile`
> 
> (in order to unblock you)
> 
> I've just tried one submission and it seems to work

Thanks, that worked for me (this week! 😅)

lacostej commented:
Question: what happens if you do not update your Deliverfile? Does fastlane fail because the setting doesn't exist?

Has this been tested?

fr0l commented:
> Question: what happens if you do not update your Deliverfile? Does fastlane fail because the setting doesn't exist?
> 
> Has this been tested?

In this PR i've tested that if `Deliverfile` still contains key `add_id_info_uses_idfa`, the _fastlane_ will ignore it and will not explode/fail.
Maybe it's good to throw a warning at some point about "unsupported" setting (they key `add_id_info_uses_idfa`)..

angelolloqui commented:
I am afraid removing the parameter is not helping. When I get rid of it then I get another error complaining about the IDFA not present but required. I am blocked because no matter what I do it fails. By the way, removing it was fine in las build, but somehow it does no longer work for us. Anyone else with a similar issue?

fr0l commented:
> I am afraid removing the parameter is not helping. When I get rid of it then I get another error complaining about the IDFA not present but required. I am blocked because no matter what I do it fails. By the way, removing it was fine in las build, but somehow it does no longer work for us. Anyone else with a similar issue?

Hey @angelolloqui please could you check if setting param to `false` helps?

angelolloqui commented:
Hey @fr0l , I can try, but it does not make much sense to me putting it to false since we are really using IDFA, so that would basically be "tricking" right? Isn't the real problem here that Apple has changed the API and we need to adapt to the newest?

fr0l commented:
> Hey @fr0l , I can try, but it does not make much sense to me putting it to false since we are really using IDFA, so that would basically be "tricking" right? Isn't the real problem here that Apple has changed the API and we need to adapt to the newest?

Hey @angelolloqui , 
It seems that [App Store Connect APIs for declaring IDFA usage are now removed](https://developer.apple.com/documentation/appstoreconnectapi/app_store_connect_api_release_notes/app_store_connect_api_2_0_release_notes).
And now it is done on a device itself by obtaining the user’s permission using [App Tracking Transparency](https://developer.apple.com/documentation/apptrackingtransparency).

Although the release notes also say
> You may also need to provide information about IDFA usage in your app’s privacy details on the App Store

I'll look into that and maybe it's good to reuse the key `add_id_info_uses_idfa`

rogerluan commented on fastlane/swift/Fastlane.swift:
Could you revert all changes made to this .swift file? It's an autogenerated file that will be updated during a deployment 🙏

fr0l replied:
Oh. will do! Thank you!

fr0l replied:
Hey @rogerluan , done! Reverted the changes to `fastlane/swift/Fastlane.swift`.
I presume `fastlane/swift/DeliverfileProtocol.swift` is ok , or should I revert changes there too? 
Also maybe good to add to header comment `// autogenerated` ? :-)

rogerluan reviewed (dismissed):
Just a minor change request. Thanks for getting this fixed!

fr0l committed:
[fastlane] Revert changes to fastlane/swift/Fastlane.swift

fr0l commented:
The [build failing ](https://app.circleci.com/pipelines/github/fastlane/fastlane/6410/workflows/fded21f5-c613-47c5-bd0e-e872780608f9/jobs/99350/parallel-runs/0/steps/0-112)

with the test from `fastlane_core/spec/command_executor_spec.rb:11` - does not seem like any related changes. On local environment the result is `OK` for this test. Xcode 15.3 and MRI Ruby 3.3

Is it possible to re-run the build?

fr0l commented:
> > Hey @fr0l , I can try, but it does not make much sense to me putting it to false since we are really using IDFA, so that would basically be "tricking" right? Isn't the real problem here that Apple has changed the API and we need to adapt to the newest?
> 
> Hey @angelolloqui , It seems that [App Store Connect APIs for declaring IDFA usage are now removed](https://developer.apple.com/documentation/appstoreconnectapi/app_store_connect_api_release_notes/app_store_connect_api_2_0_release_notes). And now it is done on a device itself by obtaining the user’s permission using [App Tracking Transparency](https://developer.apple.com/documentation/apptrackingtransparency).
> 
> Although the release notes also say
> 
> > You may also need to provide information about IDFA usage in your app’s privacy details on the App Store
> 
> (which looks like "optional" because of `may also`). I'll look into that and maybe it's good to reuse the key `add_id_info_uses_idfa`

Hey @angelolloqui ,
I've looked around [the OpenAPI specs](https://developer.apple.com/documentation/appstoreconnectapi)
And haven't found anything close to the **App Privacy** **Product Page Preview**.
I.e. seems that whatever it is specified there - is not available using OpenAPI. And is set up once for each app.
Here is an example:

<img width="1396" alt="Screenshot 2024-05-25 at 00 14 38" src="https://github.com/fastlane/fastlane/assets/3276590/cd8d5b03-9e69-4a1b-b59e-9b2bdec093c1">

<img width="1382" alt="Screenshot 2024-05-25 at 00 20 21" src="https://github.com/fastlane/fastlane/assets/3276590/4b4a27e9-f31d-4f5e-a322-c0cc91d78ec5">

So I guess in this PR it is just deletion of the functionality that was removed from OpenAPI.

So the key `add_id_info_uses_idfa` in `Deliverfile` is becoming unused, and it is good to delete it from the Deliverfile after merging the PR.

I'm not sure - what would be good option here - probably to show warning that `add_id_info_uses_idfa` is deprecated and needs to be removed.

angelolloqui commented:
Ok thanks! With your info a tried to put it to false and it worked this time, so I think you are right it must be removed

EthanSK commented:
Can this be merged? Its quite annoying having to manually go and add my changes for review. Android is fully automated, iOS is still the pain point

SomeRandomiOSDev commented:
I too am waiting on this change as our submission pipeline is currently broken due to this issue. Is it possible to get this merged through and released?

AliSoftware commented:
@fr0l  Can you update your fork to the latest `master` and merge it back to this PR (or rebase this PR on top of the latest `fastlane:master`?

One of the CI failure that we have on this PR and that prevents it from being merged is that the `Validate Fastlane.swift generation` CI job is failing because it's still using an image with Xcode `12.5.1`… which has been updated and fixed in `fastlane:master` very recently; so rebasing / upmerging `master` should solve that test failure magically 🙂

fr0l commented:
> @fr0l Can you update your fork to the latest `master` and merge it back to this PR (or rebase this PR on top of the latest `fastlane:master`?
> 
> One of the CI failure that we have on this PR and that prevents it from being merged is that the `Validate Fastlane.swift generation` CI job is failing because it's still using an image with Xcode `12.5.1`… which has been updated and fixed in `fastlane:master` very recently; so rebasing / upmerging `master` should solve that test failure magically 🙂

Hey @AliSoftware . done! rebased on latest master.

mollyIV reviewed (approved):
Thank you for making this PR. It looks good to me 🚀

SomeRandomiOSDev commented:
@AliSoftware is there a process for requesting a release with this change?

AliSoftware commented:
> @AliSoftware is there a process for requesting a release with this change?

@joshdholtz just [recently worked on automating the release process for new fastlane versions](https://mastodon.social/@joshdholtz/112637615677467343). They are not yet completely automated by scheduled job yet (he'll still trigger the automation himself while he fixes any tweak in that automation) but very soon should be.

joshdholtz commented:
I'll kick one of now so I can do some more tweaking on it 😊 Thanks for the ping!

SomeRandomiOSDev commented:
Thank you, @AliSoftware and @joshdholtz!! 🙂
